06-building-a-program-start-to-finish/writing-tests/output/spec.js

```diff
@@ -0,0 +1,83 @@
+
+const invertedIndex = new InvertedIndex();
+const mockData = [
+  {
+    title: 'Alice in Wonderland',
+    text: 'Alice falls rings.'
+  },
+
+  {
+    title: 'The Lord of the Rings: The Fellowship of the Ring.',
+    text: 'An unusual alliance of in the rings.'
+  }
+];
+
+
+describe('Inverted Index Test', () => {
+  describe('It Validate Book Data', () => {
+    it('should return invalid file type if book is not an array of objects',
+     () => {
+       expect(invertedIndex.validateInput('rowland')).toBe(false);
+       expect(invertedIndex.validateInput(2)).toBe(false);
+       expect(invertedIndex.validateInput([1, 2, 3])).toBe(false);
+       expect(invertedIndex.validateInput([{ sex: 'male', age: 18 }]))
+       .toBe(false);
+       expect(invertedIndex.validateInput([{ title: 67, text: 18 }]))
+       .toBe(false);
+       expect(invertedIndex.validateInput(mockData)).toBe(true);
+     });
+  });
+  describe('Returns a concatenated string', () => {
+    it(`an object with a title and text property
+    returns a string concatenating those two properties`,
+     () => {
+       const data = { title: 'Alice in wonderland',
+         text: 'how did she get there abeg ?' };
+       expect(invertedIndex.getBookAsText(data))
+       .toBe('Alice in wonderland how did she get there abeg ?');
+       expect(invertedIndex.getBookAsText([{ sex: 'male', age: 18 }]))
```

Comment:
I think this test can be moved to another test suite. Something like, 'does not return a concatenated string for an object not containing a title and text property' i.e an invalid object.